app/src/main/java/com/moonpig/mvisample/productdetail/ProductDetailActivity.kt

```diff
@@ -42,14 +45,16 @@ class ProductDetailActivity : BaseActivity(), ProductDetailView {
                 .addToDisposables()
 
         viewModel.bindIntents(Observable.mergeArray(
-            initialIntent()
+                initialIntent()
         ))
     }
 
     private fun renderStateToView(viewState: ProductDetailScreenViewState) =
             productDetailRenderer.render(this, viewState)
 
-    private fun initialIntent() = Observable.just(ProductDetailIntent.Initial)
+    private fun initialIntent() = Observable.just(ProductDetailIntent.Initial(
+            intent.getStringExtra(productIdKey) ?: ""
```

Comment:
What happens if the empty string is emitted?

Reply:
it would be treated as an error